kernel/src/common/math.rs

```diff
@@ -49,6 +50,27 @@ pub fn closest_power_of_two(mut num: u32) -> u32 {
     num
 }
 
+#[derive(Copy,Clone,PartialEq,PartialOrd,Eq,Ord)]
+pub struct PowerOfTwo(u32);
+
+impl PowerOfTwo {
+    pub fn bit<R>(self) -> R
+        where R: From<u32>
+    {
+        From::from(self.0)
+    }
+
+    pub fn zero() -> PowerOfTwo {
+        PowerOfTwo(0)
+    }
+}
+
+impl<F: Into<u32>> From<F> for PowerOfTwo {
+    fn from(f: F) -> PowerOfTwo {
+        PowerOfTwo(log_base_two(f.into()))
```

Comment:
I think in the particular use case (of the MPU) we want either _exact_ power of two or floor, ceiling is more likely to be open to attack (e.g. overexpose memory) while floor is more likely to crash an otherwise fine process (e.g. underexpose memory).

Could just have explicit `floor` and `ceiling` functions, in which case we would probably replace the calls to `closest_power_of_two` and `from` with just `ceiling`.